webbpsf/wfirst.py

```diff
@@ -342,35 +502,56 @@ def _validate_config(self, **kwargs):
         This mainly consists of selecting the masked or unmasked pupil
         appropriately based on the wavelengths requested.
         """
-        if self.auto_pupil and self.pupil in (self._unmasked_pupil_path, self._masked_pupil_path):
-            # Does the wavelength range fit entirely in an unmasked filter?
-            wavelengths = np.array(kwargs['wavelengths'])
-            wl_min, wl_max = np.min(wavelengths), np.max(wavelengths)
-            # test shorter filters first; if wl range fits entirely in one of them, it's not going
-            # to be the (masked) wideband filter
-            if wl_max <= self.UNMASKED_PUPIL_WAVELENGTH_MAX:
-                # use unmasked pupil optic
-                self.pupil = self._unmasked_pupil_path
-                _log.info("Using the unmasked WFI pupil shape based on wavelengths requested")
-            else:
-                if wl_max > self.MASKED_PUPIL_WAVELENGTH_MAX:
-                    _log.warn("Requested wavelength is > 2e-6 m, defaulting to masked pupil shape")
-                # use masked pupil optic
-                self.pupil = self._masked_pupil_path
-                _log.info("Using the masked WFI pupil shape based on wavelengths requested")
-        else:
-            # If the user has set the pupil to a custom value, let them worry about the
-            # correct shape it should have
-            pass
+        self._pupil_controller.validate_pupil(self.filter, **kwargs)
         super(WFI, self)._validate_config(**kwargs)
 
+    @property
+    def detector(self):
+        """Detector selected for simulated PSF
+        Used in calculation of field-dependent aberrations. Must be
+        selected from detectors in the `detector_list` attribute.
+        """
+        return self._detector
```

Comment:
I believe this `detector` function is identical to that in the superclass, so the definition doesn't need to be repeated here.  In a subclass, you can override only a property setter without having to redefine the getter function, by accessing the property in the parent class with the class name. 

In other words, just below here you can change 
```
    @detector.setter
```
into
```
    @SpaceTelescopeInstrument.detector.setter
```
and not have to define `@detector` redundantly here.